api/config/v1alpha1/envoygateway_types.go

```diff
@@ -135,6 +141,58 @@ type RateLimitRedisSettings struct {
 	URL string `json:"url"`
 }
 
+// Extension defines the configuration for registering an extension to
+// the Envoy Gateway control plane.
+type Extension struct {
+	// Resources defines the set of K8s resources the extension will handle.
+	//
+	// +optional
+	Resources []GroupVersionKind `json:"resources,omitempty"`
+
+	// Hooks defines the set of hooks the extension supports
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	Hooks []ExtensionHook `json:"hooks,omitempty"`
```

Comment:
should this be tiered - as in define `XDSVirtualHost` under `post` under `xdsServer` ?

Reply:
if you think that's more readable then sure though I think the only benefit in that is a very tiny small bump in look ups? but I can't imagine the number of eventual hooks to be so large it would make a meaningful difference

Reply:
@arkodg so looking into this further, making it tiered is sorta more cumbersome to do validation on the hooks that are registered and I'm not sure what significant benefit we get from making it tiered but let me know what you think.

Reply:
the way I was thinking about this was -
* EG has a few components/runners (provider, xds-server etc)
* Each one subscribes and publishes a data type
* hooks can be added before or after to modify the data right before its published for the next stage (post) or right after subscription before ingesting the data (pre)